internal/terraform/eval_import.go

```diff
@@ -28,11 +28,6 @@ func evaluateImportIdExpression(expr hcl.Expression, target addrs.AbsResourceIns
 		})
 	}
 
-	diags = diags.Append(validateImportSelfRef(target.Resource.Resource, expr))
```

Comment:
`validateImportSelfRef` is now being done during validation, so we can remove the call along with the extra `target` argument